### ISSUE TITLE
[Feature Request] Support lazy initial state

### ISSUE THREAD

**Issue by juan-foobar** [foobaragency/foo-state#110]
## Description
It would be nice to be able to set the initial state lazily. This is particularly useful in case we have a heavy/expensive computation to calculate the initial state and we want to avoid running it every time the component re-renders.

This would also provide a more intuitive API as it would closer match the [useState lazy initial state functionality provided by React](https://reactjs.org/docs/hooks-reference.html#lazy-initial-state).

## Example
```typescript
const pleaseRunThisOnlyOnce = () => { /* ...someExpensiveComputation... */ }
const { useGlobalState } = createGlobalState(pleaseRunThisOnlyOnce)
```

**Comment by l-Leniac-l** [comment]
I love this idea. Maybe we can use the already existing check for accepting callbacks in the `setState` as the base for this. WDYT?

**Comment by juan-foobar** [comment]
Closed by #122.
